File: selection_sort/ruby/selection_sort.rb
def selection_sort(arr)
  n = arr.length
  # One by one move boundary of unsorted subarray
  for i in 0...n-1
    # Find the minimum element in the unsorted array
    min_index = i
    for j in (i+1)...n
      if arr[j] < arr[min_index]
        min_index = j
      end
    end
    # Swap the found minimum element with the first element
    arr[i], arr[min_index] = arr[min_index], arr[i]
  end
  arr
end

# Example usage:
array = [64, 25, 12, 22, 11]
sorted_array = selection_sort(array)
puts "Sorted array: #{sorted_array}"

File: selection_sort/go/selection_sort.go
package main

import "fmt"

// Function to perform Selection Sort
func selectionSort(arr []int) {
    n := len(arr)
    for i := 0; i < n-1; i++ {
        // Find the minimum element in the unsorted part of the array
        minIdx := i
        for j := i + 1; j < n; j++ {
            if arr[j] < arr[minIdx] {
                minIdx = j
            }
        }

        // Swap the found minimum element with the first element
        arr[i], arr[minIdx] = arr[minIdx], arr[i]
    }
}

func main() {
    // Example array
    arr := []int{64, 25, 12, 22, 11}
    fmt.Println("Original array:", arr)
    
    // Sorting the array using selection sort
    selectionSort(arr)
    fmt.Println("Sorted array:", arr)
}
